perma_web/tasks/dev.py

```diff
@@ -701,3 +707,496 @@ def populate_folder_cached_path(ctx, batch_size=500):
+
+                if match := re.search(r"\n*Original registrar: (?P<registrar_id>\d+)", user.notes):
+                    user.registrar_id = int(match.group('registrar_id'))
+                    user.remove_line_from_notes('Original registrar')
+
+                if match := re.search(r"\n*Original orgs: (?P<org_ids>.+)", user.notes):
+                    orgs = [int(oid) for oid in match.group('org_ids').split(', ')]
+                    user.organizations.add(*orgs)
+                    user.remove_line_from_notes('Original orgs')
+
+                user.prepend_to_notes(f'Previously merged with user { from_user_id }.')
+                user.remove_line_from_notes('Original email')
+                to_reverse.append(user)
+
+                from_user = LinkUser.objects.get(id=from_user_id)
+                from_user.prepend_to_notes(f"Extracted previously merged user { user.id }.")
+                from_user.save(update_fields=['notes'])
+                from_users.add(from_user)
+
+        LinkUser.objects.bulk_update(to_reverse, ['email', 'is_active', 'link_count', 'registrar_id', 'notes'])
+        for user in from_users:
+            user.refresh_from_db()
+            if match := re.search(r"\n*Added registrar during the merging of accounts", user.notes):
+                user.registrar_id = None
+                user.remove_line_from_notes('Added registrar')
+
+            if match := re.search(r"\n*Added organizations.*: (?P<org_ids>.+)", user.notes):
+                orgs = [int(oid) for oid in match.group('org_ids').split(', ')]
+                user.organizations.remove(*orgs)
+                user.remove_line_from_notes('Added organizations')
+
+            user.link_count = user.created_links.count()
+            user.save(update_fields=['link_count', 'registrar_id', 'notes'])
+
+        print(reversed_org_links, reversed_personal_links, to_reverse, from_users)
+
+
+def merge_users_with_only_unconfirmed_accounts(user_list):
+    """
+    Sync all the registrars/orgs to the most recently created and delete the others.
+    """
+    user_list.sort(key=lambda u: u.id, reverse=True)
+    to_keep, *to_delete = user_list
+    merge_accounts(to_keep, to_delete)
+
+
+def merge_users_with_only_one_confirmed_account(user_list):
+    """
+    Sync all the registrars/orgs to the confirmed one and delete the other ones.
+    """
+    user_list.sort(key=lambda u: u.is_confirmed, reverse=True)
+    to_keep, *to_delete = user_list
+    merge_accounts(to_keep, to_delete)
+
+
+def merge_users_with_multiple_confirmed_accounts_but_no_links(user_list):
+    """
+    Select the account they have logged into most recently, or if they
+    have never logged in, the most recently created confirmed account.
+    Then sync all the registrars/orgs to it and delete the other ones.
+    """
+    to_keep = None
+    to_delete = []
+
+    if all(u.last_login for u in user_list):
```

Comment:
This is wrong. This keeps the account they have logged into most recently only if they have logged into all their accounts; else, goes with the most recent. I think it wants to be `any`, rather than `all`. Give it another think later with a fresh brain.